src/styled.js

```diff
@@ -14,15 +18,23 @@ const setup = (pragma, prefix, forward, theme) => {
  */
 function styled(tag) {
     const _ctx = this || {};
-    return function() {
+
+    return function wrapper() {
         const _args = arguments;
 
         function Styled(props, ref) {
+            // Grab a shallow copy of the props
+            // _ctx.p: is the props sent to the context
             _ctx.p = Object.assign({ theme: useTheme && useTheme() }, props);
+
+            // Set a flag if the current components had a previous className
+            // similar to goober. This is the append/prepend flag
             _ctx.o = /\s*go[0-9]+/g.test(_ctx.p.className);
+
+            // Define the new className
             return h(
                 tag,
-                Object.assign({}, props, {
+                Object.assign(_ctx.p, {
```

Comment:
Ctx.p potentially has a theme in it so we would switch the props reference. This doesn’t seem ideal/deterministic

Reply:
That's a good catch. Thank you! Size is the same 😃